pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -433,24 +427,21 @@ stdenv.mkDerivation ({
     for p in "''${pkgsBuildBuild[@]}" "''${pkgsBuildHost[@]}" "''${pkgsBuildTarget[@]}"; do
       ${buildPkgDb nativeGhc "$setupPackageConfDir"}
     done
-    ${nativeGhcCommand}-pkg --${nativePackageDbFlag}="$setupPackageConfDir" recache
+    ${nativeGhcCommand}-pkg --package-db="$setupPackageConfDir" recache
   ''
   # For normal components
   + ''
-    for p in "''${pkgsHostHost[@]}" "''${pkgsHostTarget[@]}"; do
-      ${buildPkgDb ghc "$packageConfDir"}
-      if [ -d "$p/include" ]; then
-        configureFlags+=" --extra-include-dirs=$p/include"
-      fi
-      if [ -d "$p/lib" ]; then
-        configureFlags+=" --extra-lib-dirs=$p/lib"
-      fi
-    ''
-    # It is not clear why --extra-framework-dirs does work fine on Linux
-    + optionalString (!stdenv.buildPlatform.isDarwin || versionAtLeast nativeGhc.version "8.0") ''
-      if [[ -d "$p/Library/Frameworks" ]]; then
-        configureFlags+=" --extra-framework-dirs=$p/Library/Frameworks"
-      fi
+for p in "''${pkgsHostHost[@]}" "''${pkgsHostTarget[@]}"; do
+  ${buildPkgDb ghc "$packageConfDir"}
+  if [ -d "$p/include" ]; then
+    configureFlags+=" --extra-include-dirs=$p/include"
+  fi
+  if [ -d "$p/lib" ]; then
+    configureFlags+=" --extra-lib-dirs=$p/lib"
+  fi
+if [[ -d "$p/Library/Frameworks" ]]; then
+  configureFlags+=" --extra-framework-dirs=$p/Library/Frameworks"
+fi
```

Comment:
I assume that happened because of the common whitespace prefix elimination and you likely indented the last part which you added to this string to little. If you indent this complete string uniformly this should work.